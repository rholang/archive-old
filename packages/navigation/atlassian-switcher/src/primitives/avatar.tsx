import * as React from 'react';
import styled from 'styled-components';
import { gridSize } from '@atlaskit/theme';

interface Props {
  avatarUrl: string | null;
  fallbackComponent: React.ReactNode;
}

interface State {
  imageLoadFailed: boolean;
}

const ImageContainer = styled.img`
  height: ${gridSize() * 3}px;
  width: ${gridSize() * 3}px;
  margin: ${gridSize() / 2}px;
  border-radius: 3px;
  position: absolute;
  top: 0;
`;

const Container = styled.div`
  position: relative;
`;

export default class Avatar extends React.Component<Props, State> {
  state = {
    imageLoadFailed: false,
  };

  render() {
    const { avatarUrl, fallbackComponent } = this.props;
    return (
      <Container>
        {fallbackComponent}
        {avatarUrl && !this.state.imageLoadFailed && (
          <ImageContainer src={avatarUrl} onError={this.onError} />
        )}
      </Container>
    );
  }

  private onError = () => {
    this.setState({ imageLoadFailed: true });
  };
}
