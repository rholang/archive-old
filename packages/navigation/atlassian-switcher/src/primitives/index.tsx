export { default as SwitcherThemedItem } from './themed-item';
export {
  default as SwitcherThemedItemWithEvents,
} from './themed-item-with-events';
export { default as SwitcherThemedChildItem } from './themed-child-item';
export { default as SwitcherItem } from './item';
export { default as SwitcherItemWithDropdown } from './item-with-dropdown';
export { default as Section } from './section';
export { default as SwitcherWrapper } from './wrapper';
export { default as ManageButton } from './manage-button';
export { default as Skeleton } from './skeleton';
export { default as TryLozenge } from './try-lozenge';
export { default as FormattedMessage } from './formatted-message';
