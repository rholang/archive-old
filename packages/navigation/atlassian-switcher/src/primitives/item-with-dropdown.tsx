import * as React from 'react';
import styled from 'styled-components';
import ChevronDownIcon from '@atlaskit/icon/glyph/chevron-down';
import ChevronUpIcon from '@atlaskit/icon/glyph/chevron-up';
import { Toggle, ItemWrapper } from './themed-top-level-item';
import SwitcherThemedItem from './themed-item';
import SwitcherThemedChildItem from './themed-child-item';
import { gridSize } from '@atlaskit/theme';
import * as colors from '@atlaskit/theme/colors';
import Tooltip from '@atlaskit/tooltip';
import Avatar from './avatar';
import { FadeIn } from './fade-in';
import { SwitcherChildItem } from '../types';

import {
  createAndFireNavigationEvent,
  withAnalyticsEvents,
  UI_EVENT_TYPE,
  SWITCHER_CHILD_ITEM_SUBJECT,
  SWITCHER_ITEM_SUBJECT,
  SWITCHER_ITEM_EXPAND_SUBJECT,
  WithAnalyticsEventsProps,
} from '../utils/analytics';

const ItemContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;
  border-radius: 3px;
`;

const ChildItemsContainer = styled.div`
  margin: 2px 0;
  border-radius: 3px;
  background-color: ${colors.N20A};
`;

const gridSizeResult = gridSize();
const IconWrapper = styled.div`
  width: ${gridSizeResult * 4}px;
  height: ${gridSizeResult * 4}px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: -${gridSizeResult}px;
  margin-bottom: -1px;
`;

interface Props extends WithAnalyticsEventsProps {
  children: React.ReactNode;
  icon: React.ReactNode;
  tooltipContent: React.ReactNode;
  description?: React.ReactNode;
  onChildItemClick?: Function;
  onExpandClick?: Function;
  onItemClick?: Function;
  href?: string;
  isDisabled?: boolean;
  childIcon?: React.ReactNode;
  childItems?: SwitcherChildItem[];
}

interface State {
  itemHovered: boolean;
  showChildItems: boolean;
}

class SwitcherItemWithDropDown extends React.Component<Props, State> {
  state = {
    itemHovered: false,
    showChildItems: false,
  };

  render() {
    const {
      icon,
      description,
      childItems,
      childIcon,
      onItemClick,
      onChildItemClick,
      ...rest
    } = this.props;
    const { showChildItems, itemHovered } = this.state;
    const childItemsExist = childItems && childItems.length > 0;

    return (
      <FadeIn>
        <React.Fragment>
          <ItemContainer
            onMouseEnter={this.onMouseEnter}
            onMouseLeave={this.onMouseLeave}
          >
            <ItemWrapper isParentHovered={itemHovered}>
              <SwitcherThemedItem
                icon={icon}
                description={childItemsExist ? description : null}
                onClick={onItemClick}
                {...rest}
              />
            </ItemWrapper>
            {childItemsExist && this.renderToggle(showChildItems, itemHovered)}
          </ItemContainer>
          {showChildItems && childItems && (
            <ChildItemsContainer>
              {childItems.map(item => (
                <SwitcherThemedChildItem
                  icon={
                    <Avatar
                      avatarUrl={item.avatar}
                      fallbackComponent={childIcon}
                    />
                  }
                  href={item.href}
                  key={item.label}
                  onClick={onChildItemClick}
                  data-test-id="switcher-child-item"
                >
                  {item.label}
                </SwitcherThemedChildItem>
              ))}
            </ChildItemsContainer>
          )}
        </React.Fragment>
      </FadeIn>
    );
  }

  private renderToggle(showChildItems: boolean, isParentHovered: boolean) {
    const icon = (
      <IconWrapper>
        {showChildItems ? (
          <ChevronUpIcon label="Collapse" />
        ) : (
          <ChevronDownIcon label="Expand" />
        )}
      </IconWrapper>
    );

    const toggle = (
      <Toggle isParentHovered={isParentHovered}>
        <SwitcherThemedItem
          data-test-id="switcher-expand-toggle"
          onClick={this.toggleChildItemsVisibility}
          onKeyDown={(e: KeyboardEvent) =>
            e.key === 'Enter' && this.toggleChildItemsVisibility()
          }
          children={undefined}
          icon={icon}
        />
      </Toggle>
    );

    return showChildItems ? (
      toggle
    ) : (
      <Tooltip
        content={this.props.tooltipContent}
        hideTooltipOnMouseDown
        position="top"
      >
        {toggle}
      </Tooltip>
    );
  }

  private toggleChildItemsVisibility = (event?: React.SyntheticEvent) => {
    event && event.preventDefault();
    this.setState({
      showChildItems: !this.state.showChildItems,
    });

    if (!this.state.showChildItems) {
      this.props.onExpandClick && this.props.onExpandClick();
    }
  };

  private setItemHovered = (value: boolean) => {
    this.setState({
      itemHovered: value,
    });
  };

  private onMouseEnter = () => this.setItemHovered(true);
  private onMouseLeave = () => this.setItemHovered(false);
}

const SwitcherItemWithDropDownWithEvents = withAnalyticsEvents({
  onChildItemClick: createAndFireNavigationEvent({
    eventType: UI_EVENT_TYPE,
    action: 'clicked',
    actionSubject: SWITCHER_CHILD_ITEM_SUBJECT,
  }),
  onExpandClick: createAndFireNavigationEvent({
    eventType: UI_EVENT_TYPE,
    action: 'clicked',
    actionSubject: SWITCHER_ITEM_EXPAND_SUBJECT,
  }),
  onItemClick: createAndFireNavigationEvent({
    eventType: UI_EVENT_TYPE,
    action: 'clicked',
    actionSubject: SWITCHER_ITEM_SUBJECT,
  }),
})(SwitcherItemWithDropDown);

export default SwitcherItemWithDropDownWithEvents;
