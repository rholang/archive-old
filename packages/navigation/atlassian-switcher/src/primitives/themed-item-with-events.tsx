import ThemedItem from './themed-item';

import {
  createAndFireNavigationEvent,
  withAnalyticsEvents,
  UI_EVENT_TYPE,
  SWITCHER_ITEM_SUBJECT,
} from '../utils/analytics';

const SwitcherItemWithEvents = withAnalyticsEvents({
  onClick: createAndFireNavigationEvent({
    eventType: UI_EVENT_TYPE,
    action: 'clicked',
    actionSubject: SWITCHER_ITEM_SUBJECT,
  }),
})(ThemedItem);

export default SwitcherItemWithEvents;
