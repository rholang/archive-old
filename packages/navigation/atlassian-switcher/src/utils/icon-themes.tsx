import * as React from 'react';
import styled from 'styled-components';
import { colors, elevation, gridSize } from '@atlaskit/theme';
import { ComponentType } from 'react';

interface IconBaseProps {
  bgColor?: string;
  iconElevation?: string;
}

const IconBase = styled.div<IconBaseProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${4 * gridSize()}px;
  height: ${4 * gridSize()}px;
  border-radius: ${gridSize()}px;
  ${({ iconElevation }) => (iconElevation ? iconElevation : '')};
  background-color: ${({ bgColor }) => bgColor}
  overflow: hidden;
`;

const ImageIconBase = styled.img`
  width: ${gridSize() * 4}px;
  height: ${gridSize() * 4}px;
`;

interface AkIconProps {
  primaryColor?: string;
  secondaryColor?: string;
  size?: 'small' | 'medium' | 'large' | 'xlarge';
}

interface AkLogoProps {
  iconGradientStart?: string;
  iconGradientStop?: string;
  iconColor?: string;
  size?: 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
}

interface IconBackgroundTheme {
  backgroundColor?: string;
}

interface IconTheme {
  primaryColor?: string;
  iconColor?: string;
  iconElevation?: string;
  iconGradientStart?: string;
  iconGradientStop?: string;
}

type IconThemeMap = {
  [index: string]: IconTheme & IconBackgroundTheme;
};

export const themes: IconThemeMap = {
  default: {
    backgroundColor: '#fff',
    primaryColor: '#000',
    iconElevation: elevation.e100,
  },
  product: {
    iconColor: colors.N0,
    backgroundColor: colors.B400,
    primaryColor: colors.N0,
    iconElevation: elevation.e100,
  },
  admin: {
    backgroundColor: colors.DN70,
    primaryColor: colors.N0,
    iconElevation: elevation.e100,
  },
  custom: {
    backgroundColor: colors.N0,
    primaryColor: colors.DN70,
    iconElevation: elevation.e100,
  },
  subtle: {
    backgroundColor: 'transparent',
    primaryColor: colors.text,
  },
  recommendedProduct: {
    backgroundColor: colors.N30,
    iconColor: colors.B200,
    iconGradientStart: colors.B400,
    iconGradientStop: colors.B200,
    iconElevation: elevation.e100,
  },
  discover: {
    backgroundColor: colors.N30,
    primaryColor: colors.DN90,
    iconElevation: elevation.e100,
  },
};

interface IconProps {
  theme: string;
}

export type IconType = ComponentType<IconProps>;

export const createIcon = (
  InnerIcon: React.ComponentType<any>,
  defaultProps?: AkIconProps | AkLogoProps,
): IconType => props => {
  const { backgroundColor, iconElevation, ...iconProps } =
    themes[props.theme] || themes.default;

  return (
    <IconBase bgColor={backgroundColor} iconElevation={iconElevation}>
      <InnerIcon {...defaultProps} {...iconProps} />
    </IconBase>
  );
};

export const createImageIcon = (url: string): IconType => props => {
  const { backgroundColor } = themes[props.theme] || themes.default;

  return (
    <IconBase bgColor={backgroundColor}>
      <ImageIconBase src={url} />
    </IconBase>
  );
};
