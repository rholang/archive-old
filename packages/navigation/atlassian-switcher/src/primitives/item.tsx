import * as React from 'react';
import Item from '@atlaskit/item';
import { WithAnalyticsEventsProps } from '../utils/analytics';
import { FadeIn } from './fade-in';

export interface SwitcherItemProps extends WithAnalyticsEventsProps {
  children: React.ReactNode;
  icon: React.ReactNode;
  description?: React.ReactNode;
  onClick?: Function;
  href?: string;
  isDisabled?: boolean;
  onKeyDown?: any;
}

export default class SwitcherItem extends React.Component<SwitcherItemProps> {
  render() {
    const { icon, description, ...rest } = this.props;
    return (
      <FadeIn>
        <Item elemBefore={icon} description={description} {...rest} />
      </FadeIn>
    );
  }
}
