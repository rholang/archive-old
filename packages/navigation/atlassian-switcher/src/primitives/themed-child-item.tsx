import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { itemThemeNamespace } from '@atlaskit/item';
import Item, { SwitcherItemProps } from './item';
import { ChildItemTheme } from '../theme/default-theme';
import { Themed } from '../theme/types';

export default (props: Themed<SwitcherItemProps>) => (
  <ChildItemTheme.Consumer>
    {tokens => (
      <ThemeProvider theme={{ [itemThemeNamespace]: tokens }}>
        <Item {...props} />
      </ThemeProvider>
    )}
  </ChildItemTheme.Consumer>
);
