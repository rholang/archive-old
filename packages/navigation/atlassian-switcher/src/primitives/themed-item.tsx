import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { itemThemeNamespace } from '@atlaskit/item';
import Item, { SwitcherItemProps } from './item';
import { ItemTheme } from '../theme/default-theme';
import { Themed } from '../theme/types';

export default (props: Themed<SwitcherItemProps>) => (
  <ItemTheme.Consumer>
    {tokens => (
      <ThemeProvider theme={{ [itemThemeNamespace]: tokens }}>
        <Item {...props} />
      </ThemeProvider>
    )}
  </ItemTheme.Consumer>
);
