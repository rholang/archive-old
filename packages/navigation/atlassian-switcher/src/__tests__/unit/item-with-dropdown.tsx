import * as React from 'react';
import { BitbucketIcon, JiraSoftwareIcon, OpsGenieIcon } from '@atlaskit/logo';
import { shallow } from 'enzyme';
import { SwitcherItemWithDropdown } from '../../primitives';
import { createIcon } from '../../utils/icon-themes';

describe('Atlassian Switcher - ItemWithDropdown', () => {
  it('renders Bitbucket snapshot correctly', async () => {
    const Icon = createIcon(BitbucketIcon, { size: 'small' });
    const props = {
      icon: <Icon theme="product" />,
      childIcon: <Icon theme="subtle" />,
      description: null,
      href: 'https://bitbucket.org',
      childItems: [],
      tooltipContent: <span>Show more sites</span>,
    };

    const wrapper = shallow(
      <SwitcherItemWithDropdown {...props}>Bitbucket</SwitcherItemWithDropdown>,
    ).dive();

    expect(wrapper).toMatchSnapshot();
  });

  it('renders single site item correctly', async () => {
    const Icon = createIcon(OpsGenieIcon, { size: 'small' });
    const props = {
      icon: <Icon theme="product" />,
      childIcon: <Icon theme="subtle" />,
      description: 'some-instance',
      href: 'https://app.opsgenie.com',
      childItems: [],
      tooltipContent: <span>Show more sites</span>,
    };

    const wrapper = shallow(
      <SwitcherItemWithDropdown {...props}>Opsgenie</SwitcherItemWithDropdown>,
    );
    wrapper.dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an item with multiple sites correctly and clicking expand shows children sites', async () => {
    const Icon = createIcon(JiraSoftwareIcon, { size: 'small' });
    const props = {
      icon: <Icon theme="product" />,
      childIcon: <Icon theme="subtle" />,
      description: 'some-instance',
      href: 'https://app.opsgenie.com',
      childItems: [
        {
          label: 'site00',
          href:
            'https://site00.atlassian.net/secure/BrowseProjects.jspa?selectedProjectType=software',
          avatar:
            'https://site-admin-avatar-cdn.staging.public.atl-paas.net/avatars/240/rocket.png',
        },
        {
          label: 'site10',
          href:
            'https://site10.atlassian.net/secure/BrowseProjects.jspa?selectedProjectType=software',
          avatar:
            'https://site-admin-avatar-cdn.stg.public.atl-paas.net/avatars/240/rocket.png',
        },

        {
          label: 'site30',
          href:
            'https://site30.atlassian.net/secure/BrowseProjects.jspa?selectedProjectType=software',
          avatar: null,
        },
      ],
      tooltipContent: <span>Show more sites</span>,
    };

    // doing a double .dive() here to render the toggle and child items
    const wrapper = shallow(
      <SwitcherItemWithDropdown {...props}>
        Jira Software
      </SwitcherItemWithDropdown>,
    )
      .dive()
      .dive();

    expect(wrapper).toMatchSnapshot();

    const expandToggle = wrapper.find(
      '[data-test-id="switcher-expand-toggle"]',
    );

    // child items are visible after the expand toggle is clicked once
    expandToggle.at(0).simulate('click');
    expect(wrapper).toMatchSnapshot();

    // child items are hidden after the expand toggle is clicked again
    expandToggle.at(0).simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
