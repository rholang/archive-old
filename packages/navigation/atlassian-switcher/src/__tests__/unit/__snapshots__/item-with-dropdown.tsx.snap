// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atlassian Switcher - ItemWithDropdown renders Bitbucket snapshot correctly 1`] = `
<SwitcherItemWithDropDown
  childIcon={
    <Unknown
      theme="subtle"
    />
  }
  childItems={Array []}
  createAnalyticsEvent={[Function]}
  description={null}
  href="https://bitbucket.org"
  icon={
    <Unknown
      theme="product"
    />
  }
  onChildItemClick={[Function]}
  onExpandClick={[Function]}
  onItemClick={[Function]}
  tooltipContent={
    <span>
      Show more sites
    </span>
  }
>
  Bitbucket
</SwitcherItemWithDropDown>
`;

exports[`Atlassian Switcher - ItemWithDropdown renders an item with multiple sites correctly and clicking expand shows children sites 1`] = `
<styled.div>
  <styled.div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Component
      isParentHovered={false}
    >
      <Component
        createAnalyticsEvent={[Function]}
        description="some-instance"
        href="https://app.opsgenie.com"
        icon={
          <Unknown
            theme="product"
          />
        }
        onClick={[Function]}
        onExpandClick={[Function]}
        tooltipContent={
          <span>
            Show more sites
          </span>
        }
      >
        Jira Software
      </Component>
    </Component>
    <WithAnalyticsContext(WithAnalyticsEvents(Tooltip))
      content={
        <span>
          Show more sites
        </span>
      }
      hideTooltipOnMouseDown={true}
      position="top"
    >
      <Component
        isParentHovered={false}
      >
        <Component
          data-test-id="switcher-expand-toggle"
          icon={
            <styled.div>
              <ChevronDownIcon
                label="Expand"
              />
            </styled.div>
          }
          onClick={[Function]}
          onKeyDown={[Function]}
        />
      </Component>
    </WithAnalyticsContext(WithAnalyticsEvents(Tooltip))>
  </styled.div>
</styled.div>
`;

exports[`Atlassian Switcher - ItemWithDropdown renders an item with multiple sites correctly and clicking expand shows children sites 2`] = `
<styled.div>
  <styled.div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Component
      isParentHovered={false}
    >
      <Component
        createAnalyticsEvent={[Function]}
        description="some-instance"
        href="https://app.opsgenie.com"
        icon={
          <Unknown
            theme="product"
          />
        }
        onClick={[Function]}
        onExpandClick={[Function]}
        tooltipContent={
          <span>
            Show more sites
          </span>
        }
      >
        Jira Software
      </Component>
    </Component>
    <Component
      isParentHovered={false}
    >
      <Component
        data-test-id="switcher-expand-toggle"
        icon={
          <styled.div>
            <ChevronUpIcon
              label="Collapse"
            />
          </styled.div>
        }
        onClick={[Function]}
        onKeyDown={[Function]}
      />
    </Component>
  </styled.div>
  <styled.div>
    <Component
      data-test-id="switcher-child-item"
      href="https://site00.atlassian.net/secure/BrowseProjects.jspa?selectedProjectType=software"
      icon={
        <Avatar
          avatarUrl="https://site-admin-avatar-cdn.staging.public.atl-paas.net/avatars/240/rocket.png"
          fallbackComponent={
            <Unknown
              theme="subtle"
            />
          }
        />
      }
      key="site00"
      onClick={[Function]}
    >
      site00
    </Component>
    <Component
      data-test-id="switcher-child-item"
      href="https://site10.atlassian.net/secure/BrowseProjects.jspa?selectedProjectType=software"
      icon={
        <Avatar
          avatarUrl="https://site-admin-avatar-cdn.stg.public.atl-paas.net/avatars/240/rocket.png"
          fallbackComponent={
            <Unknown
              theme="subtle"
            />
          }
        />
      }
      key="site10"
      onClick={[Function]}
    >
      site10
    </Component>
    <Component
      data-test-id="switcher-child-item"
      href="https://site30.atlassian.net/secure/BrowseProjects.jspa?selectedProjectType=software"
      icon={
        <Avatar
          avatarUrl={null}
          fallbackComponent={
            <Unknown
              theme="subtle"
            />
          }
        />
      }
      key="site30"
      onClick={[Function]}
    >
      site30
    </Component>
  </styled.div>
</styled.div>
`;

exports[`Atlassian Switcher - ItemWithDropdown renders an item with multiple sites correctly and clicking expand shows children sites 3`] = `
<styled.div>
  <styled.div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <Component
      isParentHovered={false}
    >
      <Component
        createAnalyticsEvent={[Function]}
        description="some-instance"
        href="https://app.opsgenie.com"
        icon={
          <Unknown
            theme="product"
          />
        }
        onClick={[Function]}
        onExpandClick={[Function]}
        tooltipContent={
          <span>
            Show more sites
          </span>
        }
      >
        Jira Software
      </Component>
    </Component>
    <WithAnalyticsContext(WithAnalyticsEvents(Tooltip))
      content={
        <span>
          Show more sites
        </span>
      }
      hideTooltipOnMouseDown={true}
      position="top"
    >
      <Component
        isParentHovered={false}
      >
        <Component
          data-test-id="switcher-expand-toggle"
          icon={
            <styled.div>
              <ChevronDownIcon
                label="Expand"
              />
            </styled.div>
          }
          onClick={[Function]}
          onKeyDown={[Function]}
        />
      </Component>
    </WithAnalyticsContext(WithAnalyticsEvents(Tooltip))>
  </styled.div>
</styled.div>
`;

exports[`Atlassian Switcher - ItemWithDropdown renders single site item correctly 1`] = `
<AnalyticsContextConsumer
  createEventMap={
    Object {
      "onChildItemClick": [Function],
      "onExpandClick": [Function],
      "onItemClick": [Function],
    }
  }
  wrappedComponentProps={
    Object {
      "childIcon": <Unknown
        theme="subtle"
      />,
      "childItems": Array [],
      "children": "Opsgenie",
      "description": "some-instance",
      "href": "https://app.opsgenie.com",
      "icon": <Unknown
        theme="product"
      />,
      "tooltipContent": <span>
        Show more sites
      </span>,
    }
  }
>
  <Component />
</AnalyticsContextConsumer>
`;
